src/app/_components/SortListbox.tsx

```diff
@@ -0,0 +1,78 @@
+'use client'
+import { Fragment, useState } from 'react'
+
+import { Listbox } from '@headlessui/react'
+import { ArrowsDownUp, CaretDown, Check } from '@phosphor-icons/react/dist/ssr'
+import clsx from 'clsx'
+
+import { Icon } from '@/components/Icon'
+
+export enum SortType {
+  Newest = 'newest',
+  Oldest = 'oldest',
+}
+
+type SortSetting = {
+  id: SortType
+  name: string
+}
+
+type SortListboxProps = {
+  onSortTypeChange: (selectedOption: SortType) => void
+}
+
+export const sortSettings: SortSetting[] = [
+  { id: SortType.Newest, name: 'Newest' },
+  { id: SortType.Oldest, name: 'Oldest' },
+]
+
+export function SortListbox({ onSortTypeChange }: SortListboxProps) {
```

Comment:
TS warned me that the `onSortTypeChange` prop is not serializable.

If we were to pass this prop from a server component, it would break. But I'm not sure if we are planning to.